lib/geoengineer/gps/constants.rb

```diff
@@ -0,0 +1,43 @@
+# Constants contains the needed information to
+class GeoEngineer::GPS::Constants
+  attr_reader :base_hash, :constants_hash
+
+  def initialize(base_hash)
+    @base_hash = base_hash
+
+    # _defaults is copied into each environment
+    @defaults_hash = @base_hash["_defaults"]
+
+    # remove all _'s
+    @constants_hash = HashUtils.remove_(@base_hash)
+
+    # the local environment overrides the values
+    @constants_hash.each_pair do |environment, vals|
```

Comment:
Could do a `@constants_hash = HashUtils.deep_merge(@defaults_hash, @constants_hash)`

Reply:
not sure we want a deep merge. This is more for an override of top level keys

Reply:
e.g. sometime you want to remove all values to make an environment equal an empty hash.